### PR TITLE
Add push rule for `m.room.server_acl`

### PR DESCRIPTION
**Pull request by richvdh**
... per https://github.com/matrix-org/matrix-spec-proposals/pull/3786.

Preview: https://pr1190--matrix-spec-previews.netlify.app

### CODE REVIEW

File: changelogs/client_server/newsfragments/1190.feature [hunk 1]
```diff
@@ -0,0 +1 @@
+Add a `.m.rule.room.server_acl` push rule to match `m.room.server_acl` events, per [MSC3786](https://github.com/matrix-org/matrix-spec-proposals/pull/3786).
```
Comment (turt2live) [inline changelogs/client_server/newsfragments/1190.feature:1]: minor consistency thing:
```suggestion
Add a `.m.rule.room.server_acl` push rule to match `m.room.server_acl` events, as per [MSC3786](https://github.com/matrix-org/matrix-spec-proposals/pull/3786).
```

**Review by turt2live — APPROVED** [review]

**Comment by SimonBrandner** [comment]
Thank you!!

File: content/client-server-api/modules/push.md [hunk 2]
```diff
@@ -520,6 +520,33 @@ Definition:
 }
 ```
 
+**`.m.rule.room.server_acl`**
```
Comment (clokep) [inline content/client-server-api/modules/push.md:523]: Should this note somewhere that it was changed in 1.4?

Comment (richvdh) [reply content/client-server-api/modules/push.md:523]: bother! I had one job...

Comment (turt2live) [reply content/client-server-api/modules/push.md:523]: I'd probably pin this one on the reviewer than the author.

Comment (turt2live) [reply content/client-server-api/modules/push.md:523]: https://github.com/matrix-org/matrix-spec/pull/1201

**Pull request merged**